### PR TITLE
feat: example material mapping using new core mapper (MM5)

### PR DESCRIPTION
**Pull request by asalzburger**
This re-establishes the vanilla material mapping using the new Core modules and a very simple `CoreMaterialMapping` algorithm (this one will be kept until the full functionality is reinstalled).

For out-of-the-box validation this is already quite good:

Screenshot 2024-04-10 at 14 45 19

### CODE REVIEW

**Comment by asalzburger** [comment]
Hi @Corentin-Allaire - this is the next one, it does not yet set up the logic of pre-sensor / post-sensor mapping, but it looks quite ok as a first shot.

**Comment by asalzburger** [comment]
ping @Corentin-Allaire

File: Examples/Scripts/Python/material_mapping_core.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    materialAssingerConfig = IntersectionMaterialAssigner.Config()
+    materialAssingerConfig.surfaces = surfaces
+    materialAssinger = IntersectionMaterialAssigner(materialAssingerConfig, loglevel)
+
+    # Accumulation setup : Binned surface material accumulater
+    materialAccumulaterConfig = BinnedSurfaceMaterialAccumulater.Config()
+    materialAccumulaterConfig.materialSurfaces = surfaces
+    materialAccumulater = BinnedSurfaceMaterialAccumulater(
+        materialAccumulaterConfig, loglevel
+    )
```
Comment (Corentin-Allaire) [inline Examples/Scripts/Python/material_mapping_core.py:65]: As I was reading this I realised something (I think). With your new setup we don't have a different mapping type for homogenous surface any more I think ? Maybe something to add to the todo list ?

File: Examples/Python/src/Geometry.cpp [hunk 2]
```diff
@@ -59,6 +59,21 @@ struct GeometryIdentifierHookBinding : public Acts::GeometryIdentifierHook {
...
+  void operator()(const Acts::Surface* surface) {
+    if (surface->surfaceMaterial() != nullptr) {
+      if (std::find(surfaces.begin(), surfaces.end(), surface) ==
+          surfaces.end()) {
+        surfaces.push_back(surface);
+      }
+    }
+  }
+};
+
```
Comment (Corentin-Allaire) [inline Examples/Python/src/Geometry.cpp:76]: Shouldn't this check for proto material instead of all type of material ?

**Review by Corentin-Allaire — COMMENTED** [review]
Looks great ! Just a couple of comments then this can go in.

File: Examples/Scripts/Python/material_mapping_core.py (hunk 1 above)
Comment (asalzburger) [reply Examples/Scripts/Python/material_mapping_core.py:65]: Homogneous will just be a 1x1 binning, and then turned into a homogenous material when the maps are finalised.

**Review by Corentin-Allaire — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [a54444fc27550432a2fe7cbacaf27e77fb6af79b]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7233967)!

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069641)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069640)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069639)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069638)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069637)||
|:green_circle: | [test_ActsConversionWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069636)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069635)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069634)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069633)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069632)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069631)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069630)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069629)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069628)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069627)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069626)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069625)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069624)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069623)||
|:red_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069622)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069621)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069620)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069619)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069618)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069617)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069616)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069615)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/38069614)||

**Pull request merged**